tests/load/locustfiles/stored.py

```diff
@@ -0,0 +1,404 @@
+        for _ in range(len(self.notification_records)):
+            self.recv_message()
+
+    def recv_message(self) -> None:
+        self.on_ws_message(self.ws, self.ws.recv())
+
+    def post_notification(self, endpoint_url: str) -> None:
+        """Send a notification to Autopush.
+
+        Args:
+            endpoint_url: A channel destination endpoint url
+        Raises:
+            ZeroStatusRequestError: In the event that Locust experiences a network issue while
+                                    sending a notification.
+        """
+        message_type: str = "notification"
+        # Prefix random message with 'TestData' to more easily differentiate the payload
+        data: str = "TestData" + "".join(
+            [
+                random.choice(string.ascii_letters + string.digits)
+                for i in range(0, random.randrange(1024, 4096, 2) - 8)
+            ]
+        )
+
+        record = NotificationRecord(send_time=time.perf_counter(), data=data)
+        self.notification_records[sha1(data.encode()).digest()] = record
+
+        with self.client.post(
+            url=endpoint_url,
+            name=message_type,
+            data=data,
+            headers=self.REST_HEADERS,
+            catch_response=True,
+        ) as response:
+            if response.status_code == 0:
+                raise ZeroStatusRequestError()
+            if response.status_code != 201:
+                response.failure(f"{response.status_code=}, expected 201, {response.text=}")
+
+    def recv(self, data: str) -> Message | None:
+        """Verify the contents of an Autopush message and report response statistics to Locust.
+
+        Args:
+            data: utf-8 data received from the server
+        """
+        recv_time: float = time.perf_counter()
+        exception: str | None = None
+        message: Message | None = None
+        message_type: str = "unknown"
+        record: Record | None = None
+        response_time: float = 0
+
+        try:
+            message_dict: dict[str, Any] = json.loads(data)
+            message_type = message_dict.get("messageType", "unknown")
+            match message_type:
+                case "hello":
+                    message = HelloMessage(**message_dict)
+                    record = self.hello_record
+                case "notification":
+                    message = NotificationMessage(**message_dict)
+                    message_data: str = message.data
+                    decode_data: str = base64.urlsafe_b64decode(message_data + "===").decode(
+                        "utf8"
+                    )
+                    record = self.notification_records.pop(
+                        sha1(decode_data.encode()).digest(), None
+                    )
+                case "register":
+                    message = RegisterMessage(**message_dict)
+                    register_chid: str = message.channelID
+                    record = next(
```

Comment:
I keep thinking that these should be dicts, since we're basically treating them like that here. I get that there may be locking issues, but I'm not really convinced that they get resolved by using a list. 

Not a blocker for this PR, but something to consider.